MSMAll/MSMAllPipeline.sh

```diff
@@ -31,693 +32,232 @@
-		log_Err "ICA+FIX HighPass setting (--high-pass=) required"
-		error_count=$(( error_count + 1 ))
-	else
-		log_Msg "ICA+FIX HighPass setting: ${p_HighPass}"
-	fi
-
-	if [ -z "${p_fMRIProcSTRING}" ]; then
-		log_Err "fMRI Proc string (--fmri-proc-string=) required"
-		error_count=$(( error_count + 1 ))
-	else
-		log_Msg "fMRI Proc string: ${p_fMRIProcSTRING}"
-	fi
-
-	if [ -z "${p_MSMAllTemplates}" ]; then
-		log_Err "MSM All Templates (--msm-all-templates=) required"
-		error_count=$(( error_count + 1 ))
-	else
-		log_Msg "MSM All Templates: ${p_MSMAllTemplates}"
-	fi
-
-	if [ -z "${p_OutputRegName}" ]; then
-		log_Err "Output Registration Name (--output-registration-name=) required"
-		error_count=$(( error_count + 1 ))
-	else
-		log_Msg "Output Registration Name: ${p_OutputRegName}"
-	fi
-
-	if [ -z "${p_HighResMesh}" ]; then
-		log_Err "High Resolution Mesh (--high-res-mesh=) required"
-		error_count=$(( error_count + 1 ))
-	else
-		log_Msg "High Resolution Mesh: ${p_HighResMesh}"
-	fi
-
-	if [ -z "${p_LowResMesh}" ]; then
-		log_Err "Low Resolution Mesh (--low-res-mesh=) required"
-		error_count=$(( error_count + 1 ))
-	else
-		log_Msg "Low Resolution Mesh: ${p_LowResMesh}"
-	fi
-
-	if [ -z "${p_InputRegName}" ]; then
-		log_Err "Input Registration Name (--input-registration-name=) required"
-		error_count=$(( error_count + 1 ))
-	else
-		log_Msg "Input Registration Name: ${p_InputRegName}"
-	fi
-
-	log_Msg "Myelin Target File: ${p_MyelinTargetFile}"
+pipedirguessed=0
+if [[ "${HCPPIPEDIR:-}" == "" ]]
+then
+    pipedirguessed=1
+    #fix this if the script is more than one level below HCPPIPEDIR
+    export HCPPIPEDIR="$(dirname -- "$0")/.."
+fi
 
-	if [ -z "${p_MatlabRunMode}" ]; then
-		log_Err "MATLAB run mode value (--matlab-run-mode=) required"
-		error_count=$(( error_count + 1 ))
-	else
-		case ${p_MatlabRunMode} in
-			0)
-				log_Msg "MATLAB run mode: ${p_MatlabRunMode} - Use compiled MATLAB"
-				if [ -z "${MATLAB_COMPILER_RUNTIME}" ]; then
-					log_Err_Abort "To use MATLAB run mode: ${p_MatlabRunMode}, the MATLAB_COMPILER_RUNTIME environment variable must be set"
-				else
-					log_Msg "MATLAB_COMPILER_RUNTIME: ${MATLAB_COMPILER_RUNTIME}"
-			  fi
-				;;
-			1)
-				log_Msg "MATLAB run mode: ${p_MatlabRunMode} - Use interpreted MATLAB"
-				;;
-			2)
-				log_Msg "MATLAB run mode: ${p_MatlabRunMode} - Use Octave"
-				;;
-			*)
-				log_Err "MATLAB run mode value must be 0, 1, or 2"
-				error_count=$(( error_count + 1 ))
-				;;
-		esac
-	fi
+source "$HCPPIPEDIR/global/scripts/newopts.shlib" "$@"
+source "$HCPPIPEDIR/global/scripts/debug.shlib" "$@"         # Debugging functions; also sources log.shlib
+g_matlab_default_mode=1
+
+#description to use in usage - syntax of parameters is now explained automatically
+opts_SetScriptDescription "implements MSM-All Registration Pipeline"
+
+#mandatory
+#general inputs
+opts_AddMandatory '--study-folder' 'StudyFolder' 'path' "folder that contains all subjects"
```

Comment:
I noticed `--path` is used in the example batch script, however, I only keep `--study-folder` in the updated version. Do we want to include both options in the new version? I am not aware of the proper usage in newopt to achieve this.

Reply:
https://github.com/Washington-University/HCPpipelines/blob/b8db803befadf5f80d1ec9004fe64b3d68bf7d25/ICAFIX/ReApplyFixMultiRunPipeline.sh#L50

Legacy flags that mean the same thing are specified on the end (warning, for an optional parameter, you MUST provide a default value before you can add legacy flags, but the empty string can be used).